cypress-io/cypress
# fix: Detect user-configured browsers

- Closes 

### User facing changelog

Adding a custom browser configuration will no longer emit a warning when trying to use that browser



### Additional details

If you specify a custom browser (either directly in `setupNodeEvents` or via a plugin) and try to use that browser via `--browser`, Cypress logs a warning

<img width="531" alt="image" src="https://user-images.githubusercontent.com/788456/185509558-b8f1acab-af63-4092-9ea7-7a1596e4ee47.png">

I expect this isn't the correct fix for this problem hence the draft status but hoping to get some feedback. The issue seems to be that the logic within `makeDataContext` only resolves the machine browsers based on the known browsers and ignores any user-configured browsers. Perhaps I'm just configuring things wrong and there's another way to do it.



### Steps to test

* Add a new browser like the Brave example on https://docs.cypress.io/guides/guides/launching-browsers#Customize-available-browsers
* Launch `npx cypress run --browser Brave` (or whatever you've called the custom browser*

Expected:
- Tests are run

Actual:
- Test are run but a warning is issued beforehand

### How has the user experience changed?


### PR Tasks
